### PR TITLE
Add notes on PL Data structures and Demos

### PR DESCRIPTION
**Pull request by Jwaxo**
Also there's this phantom other readme here? Anyway it's irrelevant in the v10.2.0 version.

### CODE REVIEW

File: particle/architecture/component-structure.md [hunk 1]
```diff
@@ -77,3 +75,28 @@ export function enable($context) {
...
+
+### component.md
+An additional readme that may describe how to use your demo, its implications, or others. By default the variables that a component requires are often documented here, in addition to in the component's _component.twig file.
+
+### component.yml
+An optional way to override the data in `/_data/data.yml`, to display something specific in your demo. These data structures are only included in their respective page's demos when rendering the demo's template, so they are safely separated from other pattern demos.
+
+Due to Particle including the [Faker plugin of Pattern Lab](https://github.com/pattern-lab/plugin-php-faker), you can use the [Node.JS Faker API](https://www.npmjs.com/package/faker) to generate random data to appear in your demos! See `/_data/data.yml` for examples.
+
+**Note:** If you would like to create demo data to be global between all patterns, add the variables to `/_data/data.yml`.
```
Comment (tjheffner) [inline particle/architecture/component-structure.md]: global data is any .yml file inside of source/_data, not just data.yml

File: particle/architecture/component-structure.md [hunk 2]
```diff
@@ -77,3 +75,28 @@ export function enable($context) {
 export default enable;
 ```
 
+## Using Demos
+
+Every component has an optional demo folder, which is automatically pulled into Pattern Lab. These files are not built to destination apps at all, so you should never do work in them that you expect to see in a final product.
```
Comment (tjheffner) [inline particle/architecture/component-structure.md]: "These files are only built for Pattern Lab, so you should never do work in them that you expect to see on a production app, like a Drupal theme"

File: particle/architecture/component-structure.md [hunk 3]
```diff
@@ -77,3 +75,28 @@ export function enable($context) {
...
+
+The Yeoman generator creates four files by default in this folder, which describe the general Phase2 way to demonstrate how to use a new component:
+
+### index.js
+Describes to Webpack the files to include when generating the Pattern Lab demos. This most likely will only need to be touched when you are creating multiple demo pages for a single component.
+
+### component.twig
+This should serve as an example twig file that shows an inclusion of the new pattern, placed with example variables and layouts so that it will appear like an intended component when viewed in the demos.
+
+The default .twig file places a standard Twig-style `include` of your new component, but you can (and should) modify this page to meet your design system better. Look at the default patters for good examples of how to create a demo template.
```
Comment (tjheffner) [inline particle/architecture/component-structure.md]: "default patters" => "default patterns"

File: particle/architecture/component-structure.md (hunk 1 above)
Comment (Jwaxo) [reply particle/architecture/component-structure.md]: Oooooh, this is good to know.

File: particle/architecture/component-structure.md (hunk 2 above)
Comment (Jwaxo) [reply particle/architecture/component-structure.md]: More succinct *and* informative.

File: particle/architecture/component-structure.md (hunk 3 above)
Comment (Jwaxo) [reply particle/architecture/component-structure.md]: Also fixed.

**Pull request merged**